src/Sylius/Bundle/WebBundle/Controller/Frontend/Account/OrderController.php

```diff
@@ -140,7 +140,7 @@ protected function findOrderOr404($number)
             throw $this->createNotFoundException('The order does not exist.');
         }
 
-        if (!$this->get('security.context')->isGranted('ROLE_SYLIUS_ADMIN') && $this->getUser()->getId() !== $order->getUser()->getId()) {
+        if (!$this->get('security.context')->isGranted('ROLE_SYLIUS_ADMIN') && !$this->getUser()->isEqualTo($order->getUser())) {
```

Comment:
It would be better to use a Voter and use isGranted to check the owner of the Order